### ISSUE TITLE
storeAuthStateInCookie config not working as expected in older Edge and IE11

### ISSUE THREAD

**Issue by nkelly75** [AzureAD/microsoft-authentication-library-for-js#2480]
>> Please follow the issue template below. Failure to do so will result in a delay in answering your question.

## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.x.x` (specifically 2.3.1 but believe also 2.4.0)
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

**Important**: Please fill in your exact version number above, e.g. `msal@1.1.3`.

## Framework
None (not using React/Angular etc.)

## Description
Using MSAL Browser and the Auth Code Flow with PKCE. I want to use sessionStorage vs. localStorage and prefer redirect to popup interaction. All of this works well in modern browsers (recent Chrome and Edge Chromium). I also need to support older browser including Edge 44.17763.831.0 and IE11. In an attempting to make progress with Edge 44.x I set storeAuthStateInCookie to true. (I could see that after the redirect from AAD back my application the necessary state had been cleared from session storage). This still didn't work properly:
- the code in ClientApplication.ts, handleRedirectResponse would get false from interactionInProgress and the sign in would not complete
- I believe I've narrowed this down to BrowserStorage.ts and the getItem function. It gets called with various values for key and the 'type' arg of CacheSchemaType.TEMPORARY. It wants to read from a cookie (based on storeAuthStateInCookie) but the StringUtils.isEmpty check returns true and getItem just returns null
- as a result the flow fails

By tweaking getItem I can see the flow working in Edge 44.x and even IE11. I made the following quick change:

```js
    getItem(key: string, type: string): string | object {
        const value = this.windowStorage.getItem(key);
        if (StringUtils.isEmpty(value)) {
            if (type === CacheSchemaType.TEMPORARY) {
                console.log("trying something different!!");
            } else {
                return null;
            }
        }
```

All this does is allow the function to continue (instead of returning null) so it later reaches the case for CacheSchemaType.TEMPORARY in the switch statement.

I'm not suggesting this is a perfect fix but the change is certainly enough for me to see a successful auth flow with these older browsers. It also doesn't break things for the modern browsers.

## Security
- [ ] Is this issue security related?

## Regression
- [ ] Did this behavior work before? 
Version:

## MSAL Configuration

```js
const msalConfig = {
    auth: {
        clientId: "...",
        authority: "https://login.microsoftonline.com/...",
        redirectUri: "http://localhost:3000"
    },
    cache: {
        cacheLocation: "sessionStorage", // This configures where your cache will be stored
        storeAuthStateInCookie: true // Set this to "true" if you are having issues on IE11 or Edge
    }
};
```

## Expected behavior
When sessionStorage doesn't work in older browsers the documented storeAuthStateInCookie setting should be allowed to use cookies. As the code in BrowserStorage.ts stands I can't see how that would work.

## Browsers/Environment
- [ ] Chrome
- [ ] Firefox
- [x] Edge (works fine in Edge Chromium but not in 44.17763.831.0)
- [ ] Safari
- [x] IE
- [ ] Other (Please add browser name here)

**Comment by jasonnutter** [comment]
I think this will be addressed in #2415 by this change: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/93a0f3b2558cb24117ca2263364dff539e2efecd/lib/msal-browser/src/cache/BrowserStorage.ts#L300-L314

cc: @sameerag

**Comment by nkelly75** [comment]
Thanks @jasonnutter. Confirming that if I build the latest from `dev` branch the changes merged from #2415 resolve this. I see that release 2.4.0 is now hosted [on the CDN](https://alcdn.msauth.net/browser/2.4.0/js/msal-browser.min.js). However, I think that precedes the merge of #2415. Do you know when we should expect the next release? 

Also, do you think it makes sense for applications using MSAL to use feature detection (perhaps something like Modernizr) to decide whether the `storeAuthStateInCookie` config is necessary in a given browser? If so any recommendations for what feature detection tests would help in that decision?
